src/System.Management.Automation/CoreCLR/CorePsPlatform.cs

```diff
@@ -204,6 +204,29 @@ internal static class CommonEnvVariableNames
 #endif
         }
 
+        /// <summary>
+        /// Get a temporary directory to use, needs to be unique to avoid collision
+        /// </summary>
+        public static string GetTemporaryDirectory()
```

Comment:
Does this need to be public?

And shouldn't it return the same directory every time it's called?